hibernate-core/src/main/java/org/hibernate/query/sqm/internal/QuerySqmImpl.java

```diff
@@ -564,6 +570,22 @@ private NonSelectQueryPlan buildUpdateQueryPlan() {
 		}
 	}
 
+	private NonSelectQueryPlan buildInsertQueryPlan() {
+		final SqmInsertStatement sqmInsert = (SqmInsertStatement) getSqmStatement();
+
+//		final String entityNameToUpdate = sqmInsert.getTarget().getReferencedPathSource().getHibernateEntityName();
+//		final EntityPersister entityDescriptor = getSessionFactory().getDomainModel().findEntityDescriptor( entityNameToUpdate );
+
+//		final SqmMultiTableMutationStrategy multiTableStrategy = entityDescriptor.getSqmMultiTableMutationStrategy();
+//		if ( multiTableStrategy == null ) {
+			return new SimpleInsertQueryPlan( sqmInsert, domainParameterXref );
+//		}
+//		else {
+			//TODO:
+//			return new MultiTableUpdateQueryPlan( sqmInsert, domainParameterXref, multiTableStrategy );
```

Comment:
@sebersole do we need a `MultiTableInsertQueryPlan`?

Reply:
Depends how we implement it.  This is different from delete and update in that those need special handling for the matching ids whereas insert would not.  But maybe it makes the impl easier.  The other option would be to simply have a single plan for insert handling and have it manage the multi-table aspect

Reply:
OK, I will have a close a look at that next and do whatever is simpler.

Reply:
Alright, so I had a look into this, and noticed that `update` is broken for attributes mapped to `@SecondaryTable`s. Which makes me question whether I can really be bothered doing the work to make `insert` work for `@SecondaryTable`s at this point, rather than doing it later...